DIFFERENCES.md

```diff
@@ -0,0 +1,82 @@
+
+Despite the presence of these differences, ACCP remains a drop-in replacement for the vast majority of Java applications and will result in no behavior differences other than improved performance.
+
+# Behavior Changes
+These differences are those most likely to be noticed by a consuming application and may require development effort to work around.
+
+## SignatureException
+The official documentation does not fully specify when the [Signature](https://docs.oracle.com/javase/8/docs/api/java/security/Signature.html) object is expected to throw a [SignatureException](https://docs.oracle.com/javase/8/docs/api/java/security/SignatureException.html).
+Having multiple different ways to reject a signature (such as `signature.verify() == false` and throwing a `SignatureException`) is an anti-pattern and we should try to avoid it.
+ACCP tries to never throw a `SignatureException` from `Signature.verify()` except when not doing so would introduce too many compatibility issues (such as with the [JCK](https://en.wikipedia.org/wiki/Technology_Compatibility_Kit#TCK_for_the_Java_platform).)
+Currently, ACCP will only throw a `SignatureException` when verifying an EDSA or DSA signature which is not properly encoded.
+In all other cases, ACCP will just return `false` from `Signature.verify()` when given an invalid signature.
+This is different from the default OpenJDK implementation which will also inspect the inner structure of RSA signatures to reject them with a `SignatureException` if they are improperly encoded.
+ACCP follows the guidance provided in [PKCS #1 section 8.2.2](https://tools.ietf.org/html/rfc8017#section-8.2.2) in that it does not parse the inner structure but instead does a binary comparison against the expected value.
+
+For this reason, regardless of whether you use ACCP or not, we recommend the following structure for signature verification:
+```java
+    Signature signatureObject = Signature.getInstance(SIGNATURE_ALGORITHM);
+    signatureObject.initVerify(publicKey);
+    signatureObject.update(messageToVerify);
+    boolean signatureValid = false;
+    try {
+        signatureValid = signatureObject.verify(signature);
+    } catch (final SignatureException ex) {
+        signatureValid = false;
+    }
+```
+
+## Elliptic Curve KeyPairGeneration by curve size
+Neither the JCE nor the default OpenJDK provider for Elliptic Curve Cryptography (SunEC) specify behavior for when `KeyPairGenerator.initialize(int keysize)` is called with an arbitrary value.
+Behavior is only fully specified for the values of 192, 224, 256, 384, and 521.
+This means that applications cannot depend on receiving a specific curve for any other value and may encounter compatibility issues should SunEC ever change its behavior or the application changes to a different JCE provider.
+ACCP removes this unspecified behavior by rejecting use of this method for all values not on the above list (for any ACCP version after 1.5.0) with an `InvalidParameterException`.
```

Comment:
In ACCP, the `KeyPairGenerator.initialize(int keysize)` method fails with an `InvalidParameterException` for `keysize` values other than 192, 224, 256, 384, and 521.